chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
> developers of triggers must assume that they are called every iteration, that restricts the usage of triggers.

I understand that you do not want to restrict the usage of trigger. To be honest, it seems strange to me that triggers are called from non-Trainer objects. because it takes Trainer as its argument. My suggestion is making the APIs more general If you want to support such usages officially.